packages/hooks/src/usePhoneNumberVerificationSessionTimer.ts

```diff
@@ -0,0 +1,51 @@
+import { useEffect, useState } from 'react';
+import { WS } from '@deriv/shared';
+import { useStore } from '@deriv/stores';
+import dayjs from 'dayjs';
+
+const usePhoneNumberVerificationSessionTimer = () => {
+    const [session_timer, setSessionTimer] = useState<number | undefined>();
+    const [should_show_session_timeout_modal, setShouldShowSessionTimeoutModal] = useState(false);
+    const { client } = useStore();
+    const { account_settings } = client;
+    const { phone_number_verification } = account_settings;
+
+    useEffect(() => {
+        WS.send({ time: 1 }).then((response: { error?: Error; time: number }) => {
+            if (response.error) return;
+
+            if (response.time && phone_number_verification?.session_timestamp) {
+                const request_in_milliseconds = dayjs(phone_number_verification?.session_timestamp * 1000);
+                const next_request = Math.round(request_in_milliseconds.diff(response.time * 1000) / 1000);
```

Comment:
Leave comments for these 2 lines to explain the functionality and the time for better understanding.